### ISSUE TITLE
Soundstream does not get better in training after update

### ISSUE THREAD

**Issue by BlackFox1197** [lucidrains/audiolm-pytorch#57]
I updated from 0.2.4 and the previous models didn't work anymore. In 0.2.4 after approx. 70,000 trainsteps an reconstruction where you could at least see the similarity between the original and the reconstruction was working. 
In the new update after ~70,000 trainsteps there  is only some improvement in the first 2 seconds, but its still only noise. After two seconds there is only one uniform signal. It also looks like it when plotting it. Im using librispeech to train and the configuration in the example (tried 1024 and 2048 codebooksize).
Im not sure if Im just using it wrong or its a bug.
The printet loss during training seems to go down very slowly, but i cant hear a difference.
Edit: add picture
![train_70000](https://user-images.githubusercontent.com/33451580/213237246-45128873-ae56-4d54-9bf7-923616ced515.png)

**Comment by lucidrains** [comment]
@BlackFox1197 djqualia says you'll need to train for longer https://github.com/lucidrains/audiolm-pytorch/issues/54

still strange that you got results much quicker in 0.2.4. the only things i had changed since then was adding local attention, as well as changing up the stft discriminator so it can work distributed

could you try setting [this](https://github.com/lucidrains/audiolm-pytorch/blob/main/audiolm_pytorch/soundstream.py#L270) to `False` and see if that helps?

**Comment by BlackFox1197** [comment]
Sorry I forgot to mention, that I was working with the 0.4.4 version of the framework. I updated, and wil try to train the 0.5.1 version. I will report back after 30,000 more steps.

**Comment by BlackFox1197** [comment]
I didn´t restart the notebook, so I still trained on 0.4.4. After updating, the saved models could not be loaded anymore, because of differences in the state_dict. 
When I deactivated loading the dict in the package, I got the following output:
![train_100000](https://user-images.githubusercontent.com/33451580/213410723-8e5e58aa-d1b1-4753-994d-24b99da7eef7.png)
Left side is the reconstruction, right side is the original.
I started a new training with the 0.5.1 version now. Lets see if that works.
My code for training is: 
`
from audiolm_pytorch import SoundStream, SoundStreamTrainer

soundstream = SoundStream(
    codebook_size = 2048,
    rq_num_quantizers = 8,
)

trainer = SoundStreamTrainer(
    soundstream,
    folder =  '/home/ckwdani/Music/libri/libri_clean',
    batch_size = 4,
    grad_accum_every = 8,         # effective batch size of 32
    data_max_length = 320 * 32,
    num_train_steps = 10001,
    save_model_every= 200,
    results_folder = '../content/02_Soundstream_5_1_new_libri_360/0_libri_with360_10000',
).cuda()
trainer.train()
`

and the code for the encoding and reconstruction is:
`
test = torchaudio.load("/home/ckwdani/Music/libri/libri_clean/train-clean-100/40/222/40-222-0000.flac")
testForward = soundstream(test[0].to("cuda"),  return_encoded = True)
Audio(soundstream.decoder(testForward[0]).detach().cpu().numpy()[0,0,:], rate=16000)
`

I hope I`m using it correctly.

**Comment by lucidrains** [comment]
@BlackFox1197 yup that looks ok, keep us posted!

**Comment by BlackFox1197** [comment]
After training for 250,000  trainsteps, it does not seem to get near the performance of the 0.2.3 version after 32,000 steps..

**This is 0.5.1 after 250,000 steps:**
https://user-images.githubusercontent.com/33451580/214311780-55e2b972-8b10-4295-85c6-ec856c5f5933.mp4
![5_1_200000](https://user-images.githubusercontent.com/33451580/214310909-c03c1219-00aa-448d-8c51-811c7887b17c.png)

**And this is 0.2.3 after 32,000 steps:**
https://user-images.githubusercontent.com/33451580/214311820-29701ba4-a1e7-40dc-b707-4f975ba397fd.mp4
![0_2_3_superb](https://user-images.githubusercontent.com/33451580/214310755-385129c1-a503-4cb2-8994-aa82cfc478eb.png)

**This is The original Audio:**
https://user-images.githubusercontent.com/33451580/214311902-bd444554-7bc5-46be-9ce3-7794560bb3b1.mp4

I want to add, that I really appreciate the work you do here, it is amazing that we can use this piece of software for our research. And I hope I do not sound demanding when spamming in this issue.

**Comment by djqualia** [comment]
FWIW--looking back at my training data, I have to agree that 0.2.3 was my best results.  At only 1000 steps in the samples were resembling the source, whereas in 0.5.1 at 1000 steps it was extremely noisy.  While 5000 steps in 0.5.1 got better, it still wasn't as good....

Could the upgrade to 0.3.x (doing what encodec does) have caused this?

**Comment by djqualia** [comment]
I believe I upgraded from 0.2.3 to 0.4.x ~1/14, and FWIW that version appears to have similarly poor results (at 1000 steps it's still very noisy).  I think this implies the degradation in training performance happened in the 0.3.x. or 0.4.x changes.

**Comment by lucidrains** [comment]
@BlackFox1197 thank you for raising this!

**Comment by jasonlevine** [comment]
Hello, sorry to reopen this issue, but has this been resolved in 0.7.9?  Yesterday I trained Soundstream using release 0.7.8 on librispeech clean for 10000 steps and the reconstructions were pitched noise that sounded nothing like the training data.  I am now retraining with 0.7.9 but based on this thread wondering if I should instead revert to 0.2.3?
Thank you!
